app/src/main/java/ch/epfl/sdp/map/MapsActivity.java

```diff
@@ -233,4 +247,13 @@ public void unDisplay(Displayable displayable) {
             displayable.unDisplayOn(Game.getInstance().getMapApi());
         });
     }
+    /**
+     * switches to a market activity, where user can buy health, shield, scan, or shrinker items
+     */
+    public void startMarket(Market backend) {
+        Log.d("MapsActivity", "start market");
+        Intent intendingMarket = new Intent(MapsActivity.this, MarketActivity.class);
+        ((MyApplication)getApplication()).appContainer.marketBackend = backend; // the only alternative is to use static setter (which they don't like)

```

Comment:
You can pass the Object via the intent's extra as showed here https://stackoverflow.com/questions/2736389/how-to-pass-an-object-from-one-activity-to-another-on-android